### PR TITLE
fix: Add executor annotations and update cluster config

### PR DESCRIPTION
**Pull request by tkyi**
- Update cluster config for SCM settings, executor stuff
- Add executor annotations to configuration overview page

### CODE REVIEW

File: docs/cluster-management/configure-api.md [hunk 1]
```diff
@@ -242,42 +246,51 @@ Set these environment variables:
...
+| SCM_SETTINGS               | Yes                       |               | JSON object with SCM settings                |
 | SECRET_OAUTH_CLIENT_ID     | Yes                       |               | Your OAuth Client Id (Application key)       |
 | SECRET_OAUTH_CLIENT_SECRET | Yes                       |               | You OAuth Client secret (Application secret) |
 | WEBHOOK_GITHUB_SECRET      | Yes for Github            |               | Secret to sign for webhooks                  |
 | SCM_GITHUB_GHE_HOST        | Yes for Github Enterprise |               | GHE host for Github Enterprise               |
-| SCM_PRIVATE_REPO_SUPPORT   | No                        | false         | Ask Github users for 'repo' scope to allow read/write access to public and private repo |
-| SCM_USERNAME               | No                        | sd-buildbot   | Username for checkout                        |
-| SCM_EMAIL                  | No                        | dev-null@screwdriver.cd | Email of user for checkout         |
+| SCM_PRIVATE_REPO_SUPPORT   | Yes for Github            | false         | Ask Github users for 'repo' scope to allow read/write access to public and private repo |
```
Comment (d2lam) [inline docs/cluster-management/configure-api.md]: This is not required: https://github.com/screwdriver-cd/scm-github/blob/master/index.js#L85

File: docs/cluster-management/configure-api.md [hunk 2]
```diff
@@ -242,42 +246,51 @@ Set these environment variables:
...
+| SCM_USERNAME               | Yes for Github            | sd-buildbot   | Username for checkout                        |
```
Comment (d2lam) [inline docs/cluster-management/configure-api.md]: These are also not required: https://github.com/screwdriver-cd/scm-github/blob/master/index.js#L88-L89

File: docs/user-guide/configuration/index.md [hunk 3]
```diff
@@ -75,6 +70,10 @@ _Note: `Parallel`, `series`, and `matrix` have not been implemented yet. Everyth
             <h4>Settings</h4>
             <p>Configurable settings for any additional build plugins added to Screwdriver.cd.</p>
         </div>
+        <div id="annotations" class="hidden">
+            <h4>Annotations</h4>
+            <p>Annotations are an array of key-value pairs, and can be either pipeline or job-level specifications. In this case, the annotation `beta.screwdriver.cd/executor` is used to designate using Docker executor to run the pipeline.</p>
```
Comment (d2lam) [inline docs/user-guide/configuration/index.md]: Annotations is an object, not array

**Review by d2lam — CHANGES_REQUESTED** [review]

**Comment by d2lam** [comment]
The rest LGTM 👍

File: docs/cluster-management/configure-api.md [hunk 4]
```diff
@@ -160,38 +160,39 @@ datastore:
...
 #### Kubernetes
 Set these environment variables:
 
 | Environment name   | Default Value | Description                                |
 |:-------------------|:--------------|:-------------------------------------------|
-| EXECUTOR_PLUGIN    |               | Set to `k8s`                               |
-| LAUNCH_VERSION     |               | Launcher version to use                    |
-| K8S_HOST           |               | Kubernetes host                            |
-| K8S_TOKEN          |               | JWT for authenticating Kubernetes requests |
+| EXECUTOR_PLUGIN    | k8s           | Default executor (eg: `k8s`, `docker`, `k8s-vm`, `jenkins`) |
```
Comment (minzcmu) [inline docs/cluster-management/configure-api.md:170]: Petey just added an executor enable field to the API. should mention that
https://github.com/screwdriver-cd/screwdriver/blob/master/config/custom-environment-variables.yaml#L64

File: docs/cluster-management/configure-api.md [hunk 5]
```diff
@@ -242,42 +246,51 @@ Set these environment variables:
...
-    github:
-        oauthClientId: YOUR-OAUTH-CLIENT-ID
-        oauthClientSecret: YOUR-OAUTH-CLIENT-SECRET
-        # Secret to add to GitHub webhooks so that we can validate them
-        secret: SUPER-SECRET-SIGNING-THING
-        # You can also configure for use with GitHub enterprise
-        # gheHost: github.screwdriver.cd
-        # Whether to support private repo
-        # privateRepo: true
+scms: {
```
Comment (minzcmu) [inline docs/cluster-management/configure-api.md]: For this, I don't think you can put json directly in local.yaml..
If they use environment variable for scmsettings, then that env `SCM_SETTINGS` can be in json

**Review by minzcmu — APPROVED** [review]
👍 LGTM

**Pull request merged**